source/client.c

```diff
@@ -1735,22 +1748,17 @@ static enum aws_mqtt_client_request_state s_pingreq_send(uint16_t message_id, bo
             return AWS_MQTT_CLIENT_REQUEST_ERROR;
         }
 
+        /* Mark down that now is when the last pingreq was sent */
+        connection->waiting_on_ping_response = true;
+
         return AWS_MQTT_CLIENT_REQUEST_ONGOING;
     }
 
     /* Check that a pingresp has been received since pingreq was sent */
-
-    uint64_t current_time = 0;
-    aws_channel_current_clock_time(connection->slot->channel, &current_time);
-
-    if (current_time - connection->last_pingresp_timestamp > connection->request_timeout_ns) {
+    if (connection->waiting_on_ping_response) {
+        connection->waiting_on_ping_response = false;
```

Comment:
What if multiple pings are in flight? Should this maybe tick up when we send, and down when we receive?

Reply:
not possible now since PING is being removed from the public API